### ISSUE TITLE
sched-simple treats PUs as cores

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#2968]
Related to #2965 and #2961, sched-simple is treating PUs as cores, which should not be the case:

```
ƒ(s=2,builddir) grondo@quartz2531:~/git/flux-core.git$ flux hwloc info
2 Machines, 72 Cores, 144 PUs
ƒ(s=2,builddir) grondo@quartz2531:~/git/flux-core.git$ flux kvs get resource.hwloc.by_rank
{"[0-1]": {"NUMANode": 2, "Package": 2, "Core": 36, "PU": 72, "cpuset": "0-71"}}
ƒ(s=2,builddir) grondo@quartz2531:~/git/flux-core.git$ flux resource list -s free -v
     STATE NNODES   NCORES    NGPUS LIST
      free      2      144        0 rank[0-1]/core[0-71]
```

Either `"cpuset"` in the `resource.hwloc.by_rank` should include only the core ids, or the object should be augmented with a `coreids` idset. Once this is done sched-simple will schedule cores instead of PUs.

**Comment by grondo** [comment]
Partial duplicate of #2854 (sorry I had lost track of this issue). However, the fix for this issue will only treat cores as cores (perhaps with fallback to old behavior), but will not fully fix #2854 because sched-simple internally will still keep cores or PUs as a flat list.
